crates/ark/src/lsp/completions/sources/composite.rs

```diff
@@ -26,6 +26,23 @@ use crate::lsp::completions::sources::CompletionSource;
 use crate::treesitter::NodeType;
 use crate::treesitter::NodeTypeExt;
 
+#[derive(Clone, Hash, PartialEq, Eq)]
+struct CompletionItemKey {
+    label: String,
+    kind_str: String,
+}
+
+impl CompletionItemKey {
+    fn new(item: &CompletionItem) -> Self {
+        Self {
+            label: item.label.clone(),
+            kind_str: item
```

Comment:
We can't just use the completion item kind here, without dealing with a hash implementation for that enum. So I decided to just use a string representation, which is convenient for logging anyway.